### DISCUSSION TITLE
Is it possible to bring back old Kaldi's chain-model style training back to Icefall

### DISCUSSION THREAD

**Discussion by davidlin409** [k2-fsa/icefall#998] [Q&A] [answered]
Recently I need to port Kaldi's chain model training to Icefall. 

Alhough Icefall's method is with better result, but we are having very not-normal setup, such that Icefall's training method (based on CTC) does not work for us. However, Icefall have lots of new algorithm, which is very hard to port one-by-one to Kaldi. 

Therefore, I am trying to use alignment file from Kaldi, to feed those alignment-data into Icefall training flow, and use chain-style training method; which is, to convert alignment (originally it is lattice in Kaldi) to training-FST-graph, and use it inside Icefall training.

However, converting alignment from Kaldi to Icefall, while putting alignment into Icefall-training-FST-graph, is very confusing. For example, I am not sure how I can turn Kaldi alignment into Icefall-training-FST graph, where original MMI-graph need to add self-eps-loop to P-gram.

Is there any documentation, or any example to port Kaldi's chain model training method back to Icefall?

PS: Chain model here is the non-E2E version of chain model in old Kaldi, not the E2E one.

**Comment by yfyeung** [comment]
Currently, there is no method.

**Reply by danpovey** [reply, marked as answer]
I think maybe because you are using a too-old recipe?  Can you try the recipe in librispeech/ASR/zipformer?
If you have enough data, e.g. a few hundred hours at least, I think it should be better than chain model.
